volatility3/framework/layers/objectstorage.py

```diff
@@ -0,0 +1,57 @@
+# This file is Copyright 2022 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import logging
+import urllib.parse
+from typing import Optional, Any, List
+
+try:
+    import s3fs
+    HAS_S3FS = True
+except ImportError:
+    HAS_S3FS = False
+
+try:
+    import gcsfs
+    HAS_GCSFS = True
+except ImportError:
+    HAS_GCSFS = False
+
+from volatility3.framework import exceptions
+from volatility3.framework.layers import resources
+
+vollog = logging.getLogger(__file__)
+
+class S3FileSystemHandler(resources.VolatilityHandler):
+    if HAS_S3FS:
```

Comment:
Hmmmm, I'd probably put this `if HAS_S3FS` outside the class definition so it's essentially having no impact.  I've never tried raising an exception inside of class rather than in a method before?  It could work, just wondering what benefit defining the class has over simply not defining it?